### PR TITLE
ref #4397 - Fix some issues in our Init calls for embedded ruby

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Found some issues during the investigation of #4397. 

Companion PR with the same changes in the test_package of conan-openstudio-ruby: https://github.com/NREL/conan-openstudio-ruby/pull/34

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/cli/main.cpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ extern "C"
 
   //void Init_ascii(); // this is not included in libenc
   void Init_big5();
+  void Init_cesu_8();
```
Comment (jmarrec) [inline src/cli/main.cpp:56]: Was missing.

File: src/cli/main.cpp [hunk 2]
```diff
@@ -222,18 +223,21 @@ int main(int argc, char* argv[]) {
...
     rb_provide("enc/big5.so");
+    Init_cesu_8();
+    rb_provide("enc/cesu_8.so");
     Init_cp949();
     rb_provide("enc/cp949.so");
     Init_emacs_mule();
     rb_provide("enc/emacs_mule.so");
     Init_euc_jp();
-    rb_provide("enc/euc_ip.so");
+    rb_provide("enc/euc_jp.so");
```
Comment (jmarrec) [inline src/cli/main.cpp:240]: Typo

File: src/cli/main.cpp [hunk 3]
```diff
@@ -339,37 +347,37 @@ int main(int argc, char* argv[]) {
     rb_provide("enc/trans/emoji_sjis_softbank.so");
 
     Init_trans_escape();
-    rb_provide("enc/trans/escape.o");
+    rb_provide("enc/trans/escape.so");
```
Comment (jmarrec) [inline src/cli/main.cpp:350]: Typo, plenty similar following.

File: src/cli/main.cpp [hunk 4]
```diff
@@ -417,6 +425,10 @@ int main(int argc, char* argv[]) {
     rb_provide("fiddle");
     rb_provide("fiddle.so");
 
+    Init_generator();
+    rb_provide("json/ext/generator");
+    rb_provide("json/ext/generator.so");
```
Comment (jmarrec) [inline src/cli/main.cpp:430]: Simple reorder alphabetically

File: src/cli/main.cpp [hunk 5]
```diff
@@ -434,17 +446,15 @@ int main(int argc, char* argv[]) {
     Init_nonblock();
     rb_provide("nonblock");
     rb_provide("nonblock.so");
+    rb_provide("io/nonblock");
+    rb_provide("io/nonblock.so");
```
Comment (jmarrec) [inline src/cli/main.cpp:450]: Consolidate in one block, instead of calling Init_nonblock() twice

File: src/cli/main.cpp (hunk 4 above)
Comment (tijcolem) [reply src/cli/main.cpp:430]: amazed these were able to slip by for so long.

**Review by tijcolem — APPROVED** [review]

**Pull request merged**